internal/internal_workflow_client.go

```diff
@@ -1167,6 +1168,8 @@ func (wc *WorkflowClient) loadCapabilities(ctx context.Context, getSystemInfoTim
 	}
 	ctx, cancel := context.WithTimeout(ctx, getSystemInfoTimeout)
 	defer cancel()
+	// Manually append the namespace to the header since the request does not have it.
+	ctx = metadata.AppendToOutgoingContext(ctx, temporalNamespaceHeaderKey, wc.namespace)
```

Comment:
@cretz I am manually setting the namespace hear here because it appears to be needed